### PR TITLE
Fix xrefs and other code warnings.

### PR DESCRIPTION
**Pull request by cgranade**
This PR fixes incorrect cross references from API documentation, and resovles misc. other code warnings. This PR also improves API documentation by hiding as `internal` functions and operations that are not part of the public API, and that were incorrectly documented as such in earlier versions.

### CODE REVIEW

File: Standard/src/AmplitudeAmplification/CommonOracles.qs [hunk 1]
```diff
@@ -13,13 +13,13 @@ namespace Microsoft.Quantum.AmplitudeAmplification {
...
     /// # See Also
-    /// - Microsoft.Quantum.Canon.ReflectionOracle
+    /// - Microsoft.Quantum.AmplitudeAmplification.ReflectionOracle
     function ReflectionStart() : ReflectionOracle {
         return ReflectionOracle(RAll0);
     }
 
     /// # Summary
-    /// Implementation of <xref:microsoft.quantum.canon.targetstatereflectionoracle>.
+    /// Implementation of <xref:Microsoft.Quantum.AmplitudeAmplification.Targetstatereflectionoracle>.
```
Comment (cgranade) [inline Standard/src/AmplitudeAmplification/CommonOracles.qs:22]:
```suggestion
    /// Implementation of <xref:Microsoft.Quantum.AmplitudeAmplification.TargetStateReflectionOracle>.
```

File: Standard/src/Preparation/UniformSuperposition.qs [hunk 2]
```diff
@@ -81,7 +81,7 @@ namespace Microsoft.Quantum.Preparation {
         let theta = ArcSin(Sqrt(IntAsDouble(2^nQubits) / IntAsDouble(nIndices)) * Sin(PI() / 6.0));
 
         ApplyToEachCA(H, targetQubits);
-        using (compareQubits = Qubit[nQubits]) {
+        use compareQubits = Qubit[nQubits] {
```
Comment (guenp) [inline Standard/src/Preparation/UniformSuperposition.qs:84]: @cgranade should there be a `{` scope here?

**Review by guenp — APPROVED** [review]
Looks good to me!

**Comment by cgranade** [comment]
Builds passed, but the problem with status reporting hit again; forcing merge.

**Pull request merged**